diff --git a/dwm.c b/dwm.c
index 4505c4a..33251c3 100644
--- a/dwm.c
+++ b/dwm.c
@@ -232,6 +232,7 @@ static void movemouse(const Arg *arg);
 static Client *nexttiled(Client *c);
 static void pop(Client *c);
 static void propertynotify(XEvent *e);
+static void pointerfocuswin(Client *c);
 static void quit(const Arg *arg);
 static Monitor *recttomon(int x, int y, int w, int h);
 static void resize(Client *c, int x, int y, int w, int h, int interact);
@@ -1085,6 +1086,7 @@ focusmon(const Arg *arg)
 	unfocus(selmon->sel, 0);
 	selmon = m;
 	focus(NULL);
+	pointerfocuswin(NULL);
 }
 
 void
@@ -1113,6 +1115,16 @@ focusstack(const Arg *arg)
 	}
 }
 
+void
+pointerfocuswin(Client *c)
+{
+    if (c) {
+        XWarpPointer(dpy, None, root, 0, 0, 0, 0, c->x + c->w / 2, c->y + c->h / 2);
+        focus(c);
+    } else
+    XWarpPointer(dpy, None, root, 0, 0, 0, 0, selmon->wx + selmon->ww / 3, selmon->wy + selmon->wh / 2);
+}
+
 Atom
 getatomprop(Client *c, Atom prop)
 {
@@ -2190,6 +2202,7 @@ togglebar(const Arg *arg)
 		XConfigureWindow(dpy, systray->win, CWY, &wc);
 	}
 	arrange(selmon);
+	pointerfocuswin(selmon->sel);
 }
 
 void
